tests/forte/pipeline_with_record_check_test.py

```diff
@@ -0,0 +1,183 @@
+        record_meta["Token"] = ["1", "2"]
+
+    def consume_next(self, pred_pack: PackType, ref_pack: PackType):
+        pred_pack_expectation = dict()
+        pred_pack_expectation["Sentence"] = ["1", "2", "3"]
+        ref_pack_expectation = dict()
+        ref_pack_expectation["Sentence"] = ["1", "2", "3"]
+        self.expected_types_and_attributes(pred_pack_expectation,
+                                           ref_pack_expectation)
+        self.check_record(pred_pack, ref_pack)
+        self.writes_record(pred_pack, ref_pack)
+
+    def get_result(self):
+        pass
+
+
+class DummyEvaluatorTwo(Evaluator):
+    """ This evaluator does nothing, just for test purpose."""
+
+    def pred_pack_record(self, record_meta: Dict[str, List[str]]):
+        record_meta["Token"] = ["1", "2"]
+
+    def consume_next(self, pred_pack: PackType, ref_pack: PackType):
+        pred_pack_expectation = dict()
+        pred_pack_expectation["Sentence"] = ["1", "2", "3"]
+        ref_pack_expectation = dict()
+        ref_pack_expectation["Document"] = ["1", "2", "3"]
+        self.expected_types_and_attributes(pred_pack_expectation,
+                                           ref_pack_expectation)
+        self.check_record(pred_pack, ref_pack)
+        self.writes_record(pred_pack, ref_pack)
+
+    def get_result(self):
+        pass
+
+
+class RecordCheckPipelineTest(unittest.TestCase):
+
+    def test_pipeline1(self):
+        """Tests reader record writing """
+
+        nlp = Pipeline[DataPack]()
+        nlp.enforce_consistency(enforce=True)
+        reader = DummySentenceReader()
+        nlp.set_reader(reader)
+        nlp.initialize()
+        data_path = data_samples_root + "/random_texts/0.txt"
+        pack = nlp.process(data_path)
+        self.assertEqual(pack._meta.record["Sentence"], ["1", "2", "3"])
+
+    def test_pipeline2(self):
+        """Tests the processor record writing"""
+
+        nlp = Pipeline[DataPack]()
+        nlp.enforce_consistency(enforce=True)
+        reader = DummySentenceReader()
+        nlp.set_reader(reader)
+        dummy = DummyPackProcessorOne()
+        nlp.add(dummy)
+        nlp.initialize()
+        data_path = data_samples_root + "/random_texts/0.txt"
+        pack = nlp.process(data_path)
+        self.assertEqual(pack._meta.record["Sentence"], ["1", "2", "3"])
+        self.assertEqual(pack._meta.record["Token"], ["1", "2"])
+        self.assertEqual(pack._meta.record["Document"], ["2"])
+
+    def test_pipeline3(self):
+        """Tests the behavior of processor raising error exception"""
+
+        nlp = Pipeline[DataPack]()
+        nlp.enforce_consistency(enforce=True)
+        reader = DummySentenceReader()
+        nlp.set_reader(reader)
+        dummy = DummyPackProcessorTwo()
+        nlp.add(dummy)
+        nlp.initialize()
+        data_path = data_samples_root + "/random_texts/0.txt"
+        with self.assertRaises(ProcessExecutionException):
+            nlp.process(data_path)
+
```

Comment:
How about we also test for `enforce=False` right after this? i.e.:

```suggestion
nlp.enforce_consistency(enforce=False)
nlp.process(data_path)
# can do some check here, or do nothing because if the exception is not thrown, it should be ok.
```